src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/IPackageAnalyzer.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.UpgradeAssistant.Dependencies;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages
+{
+    public interface IPackageAnalyzer
+    {
+        Task<bool> RunPackageAnalyzersAsync(IUpgradeContext context, IDependencyAnalysisState? analysisState, CancellationToken token);
```

Comment:
The name should be adjusted since this is not just packages, but all dependencies. Since the type says what it is maybe just `RunAsync` or `AnalyzeAsync` or something?